### PR TITLE
Issue/44 : Added Unit Test for ConfigurationSelector

### PR DESCRIPTION
**Pull request by Sandilya1795**
Please find the unit test for review.

### CODE REVIEW

File: src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    private Configuration postConfig1(){
+        Configuration configuration5 = new Configuration();
+        configuration5.setUrl("/mock/*/get/*/data");
+        configuration5.setType("POST");
+        simpleResource.setResource("mock/sample-post4.txt");
+        configuration5.setSimpleResource(simpleResource);
+        return configuration5;
+    }
+
+    private Configuration emptyConfig(){
```
Comment (Technolords) [inline src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java]: You don't need to configure an empty instance.... the logic of the class to test will simply return a null if there is no match. That is different than returning an instance (implies a match) with nothing set....

File: src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    @Test(dataProvider = DATA_SET_FOR_TEST_CONFIGURATION_SELECTION)
+    public void testConfigurationSelection(String url, Map<String ,Configuration> testConfigs, Configuration expectedConfiguration) throws IOException, JAXBException, SAXException, InterruptedException {
+        for(String key : testConfigs.keySet()) {
+            Configuration actualConfiguration = configurationSelector.findMatchingConfiguration(url, testConfigs);
+            LOGGER.info("Equals: {}", actualConfiguration.equals(expectedConfiguration));
+            Assert.assertEquals(actualConfiguration,expectedConfiguration);
+        }
+    }
+
+    private Map<String,Configuration> getConfigs() {
```
Comment (Technolords) [inline src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java]: Looks good, a nice fat map ;)

File: src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+    @DataProvider(name = DATA_SET_FOR_TEST_CONFIGURATION_SELECTION)
+    public Object[][] dataSetConfigs(){
+        return new Object[][] {
+                { "/mock/get",          getConfigs(),       getConfig1()},
+                { "/mock/100/get",      getConfigs(),       wildCardConfig()},
+                { "/mock/1/get",        getConfigs(),       getConfig2()},
+                { "/mock/1/get/0/data", getConfigs(),       wildCardConfig2()},
+                { "/mock/post",         postConfigs(),      postConfig1()}
+//                {null,         postConfigs(),      null}
```
Comment (Technolords) [inline src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java]: For an unhappy scenario, you're close... try:
```
{ "/angry/kid", getConfigs(), null}
```

File: src/main/java/net/technolords/micro/model/jaxb/resource/SimpleResource.java [hunk 4]
```diff
@@ -12,6 +12,22 @@
...
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof SimpleResource)) {
+            return false;
+        }
+        SimpleResource ref = (SimpleResource) obj;
+        return (ref.getResource().equals(this.getResource())
+//                && ref.getCachedData().equals(this.getCachedData())
```
Comment (Technolords) [inline src/main/java/net/technolords/micro/model/jaxb/resource/SimpleResource.java]: Why commenting these out? It should be complete if these are included in the equals.

**Comment by Technolords** [comment]
The test case looks good! I am happy with the changes you've made. There are a few (tiny) comments, one of them being an unhappy test case. Could you address these?

File: src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        };
+    }
+
+    @Test(dataProvider = DATA_SET_FOR_TEST_CONFIGURATION_SELECTION)
+    public void testConfigurationSelection(String url, Map<String ,Configuration> testConfigs, Configuration expectedConfiguration) throws IOException, JAXBException, SAXException, InterruptedException {
```
Comment (Technolords) [inline src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java]: Don't forget the `final` ...

**Comment by Sandilya1795** [comment]
committed the Refactored Test.

**Pull request merged**